### PR TITLE
Control home page section theme via query param

### PR DESCRIPTION
**Pull request by buberdds**
Home page needs a theme override only in a specific section (graph area).

Now with https://github.com/oasisprotocol/explorer/pull/411 there is more work to do around graph. I will create a ClickUp task to handle that part

### CODE REVIEW

**Review by csillag — APPROVED** [review]
Reviewed. LGTM.

(Tweaked it so that the default setting is hidden in the URL.)

**Review by csillag — APPROVED** [review]
Reviewed. LGTM.

(Tweaked it so that the default setting is hidden in the URL.)

File: src/app/hooks/useSearchQueryNetworkParam.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  return {
+    network: networkQueryParam as Network,
+    setNetwork: network => {
+      if (network === Network.mainnet) {
+        searchParams.delete('network')
+        setSearchParams(searchParams)
+      } else {
+        setSearchParams({ network })
+      }
```
Comment (lukaw3d) [inline src/app/hooks/useSearchQueryNetworkParam.ts:26]: Slightly more accurate:
```suggestion
      if (network === Network.mainnet) {
        searchParams.delete('network')
      } else {
        searchParams.set('network', network)
      }
      setSearchParams(searchParams)
```

(if we ever used it with other query parameters)

**Review by lukaw3d — APPROVED** [review]

File: src/app/hooks/useSearchQueryNetworkParam.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export const useSearchQueryNetworkParam = (): {
+  network: Network
+  setNetwork: (network: Network) => void
+} => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  let networkQueryParam = searchParams.get('network')
+  if (!networkQueryParam || !RouteUtils.getEnabledNetworks().includes(networkQueryParam as any)) {
+    networkQueryParam = Network.mainnet
+  }
```
Comment (lukaw3d) [inline src/app/hooks/useSearchQueryNetworkParam.ts:17]: Also?
```suggestion
  const networkQueryParam = searchParams.get('network') ?? Network.mainnet
  if (!RouteUtils.getEnabledNetworks().includes(networkQueryParam as any)) {
    throw new Error('Unexpected network')
  }
```

Comment (csillag) [reply src/app/hooks/useSearchQueryNetworkParam.ts:17]: Yes... but we have a dedicated error type to throw for invalid network, so I will use that.

**Comment by csillag** [comment]
This 40-line PR now has code from 3 devs, _and_ two approvals. I think that's enough, so I squashed and merged it.

**Pull request merged**